src/ln/channelmanager.rs

```diff
@@ -2776,6 +2792,18 @@ impl ChannelMessageHandler for ChannelManager {
 		handle_error!(self, self.internal_announcement_signatures(their_node_id, msg), their_node_id)
 	}
 
+	fn handle_channel_update(&self, msg: &msgs::ChannelUpdate) -> Result<(), HandleError> {
+		let _ = self.total_consistency_lock.read().unwrap();
+		let mut channel_state_lock = self.channel_state.lock().unwrap();
+		let channel_state = channel_state_lock.borrow_parts();
+		if let Some(channel_id) = channel_state.short_to_id.get(&msg.contents.short_channel_id) {
+			if let Some(chan) = channel_state.by_id.get_mut(channel_id) {
+				chan.channel_update(&msg);
```

Comment:
Need to check the source of the message ie their_node_id as in all the other handler functions.

Reply:
So refactored function with an internal part to catch easily MsgHandleErrInternal